app/lib/rails_env_constraint.rb

```diff
@@ -0,0 +1,18 @@
+# Used as routing constraint to expose routes only in certain environments.
+class RailsEnvConstraint
+  # @param [Array<string>] allowed_envs the environments the route(s) will be
+  #   available in.
+  def initialize(allowed_envs:)
+    @allowed_envs = allowed_envs
+  end
+
+  # Returns true if we're in an allowed env
+  #
+  # @note We always ignore the request argument since it's not used.
+  # @return [Boolean]
+  def matches?(_req = nil)
+    # NOTE: ActiveSupport::StringInquirer works with all string methods, so
```

Comment:
Fun fact: this is Liskov Substituion at work because `AS::StringInquirer` inherits from String

```ruby
[2] pry(main)> Rails.env.is_a? String
=> true
[3] pry(main)> Rails.env.class.ancestors.take(4)
=> [ActiveSupport::EnvironmentInquirer,
 ActiveSupport::StringInquirer,
 ActiveSupport::ToJsonWithActiveSupportEncoder,
 String]
```